src/pypromice/process/aws.py

```diff
@@ -91,44 +91,79 @@ def getL1(self):
         logger.info('Level 1 processing...')
         self.L0 = [addBasicMeta(item, self.vars) for item in self.L0]
         self.L1 = [toL1(item, self.vars) for item in self.L0]
-        self.L1A = reduce(xr.Dataset.combine_first, self.L1)
+
+        if self.merge_flag:
+            self.L1A = self.hard_merge(self.L1)
+        else:
+            self.L1A = reduce(xr.Dataset.combine_first, self.L1)
 
     def getL2(self):
         '''Perform L1 to L2 data processing'''
         logger.info('Level 2 processing...')
         self.L2 = toL2(self.L1A, vars_df=self.vars)
+        self.L2 = self.resample(self.L2)
```

Comment:
Yes, it will. As will the rounding performed after this. I will move all of this to the stage where we export everything in `writeArr`. Therefore resampling, rounding and reformating are performed when `writeArr` is triggered rather than as part of the processing steps.